src/terra-dev-site/app/components.2/ApplicationEventEmitter.app.mdx

```diff
@@ -0,0 +1,159 @@
+
+```js
+import ApplicationEventEmitter from 'terra-application/lib/application-event-emitter';
+```
+
+## API
+
+|method|syntax|Description|
+|---|---|---|
+|`once`|`eventEmitter.once('event-name', listener)`|Adds a one-time listener function for the event named `event-name`. The next time `event-name` is triggered, this listener is removed and then invoked.|
+|`on`|`eventEmitter.on('event-name', listener)`|Adds the listener function to the end of the listeners array for the event named `event-name`. No checks are made to see if the listener has already been added. Multiple calls passing the same combination of `event-name` and listener will result in the listener being added, and called, multiple times.|
+|`addListener`|`eventEmitter.addListener('event-name', listener)`|Alias to `on`|
+|`off`|`eventEmitter.off('event-name', listener)`|Removes the specified listener from the listener array for the event named `event-name`.|
+|`removeListener`|`eventEmitter.removeListener('event-name', listener)`|Alias to `on`|
+|`removeAllListeners`|`eventEmitter.removeAllListeners('event-name')`|Removes all listeners if not event names are specified, or those of the specified 'event-name'.|
+|`emit`|`eventEmitter.emit('event-name', arg1, arg2);`|Synchronously calls each of the listeners registered for the event named 'event-name', in the order they were registered, passing the supplied arguments to each.|
+|`eventNames`|`eventEmitter.eventNames()`|Returns an array listing the event names for which the emitter has registered listeners. The values in the array will be strings.|
+|`listenerCount`|`eventEmitter.listenerCount('event-name')`|Returns the number of listeners listening to the event named 'event-name'.|
+|`listeners`|`eventEmitter.listeners('event-name')`|Returns a copy of the array of listeners for the event named 'event-name'.|
+
+## Examples
+
+
+### `once`
+
+```js
+import ApplicationEventEmitter from 'terra-application/lib/application-event-emitter';
+
+const eventEmitter = new ApplicationEventEmitter();
+let listenCount = 0;
+const listener = () => {listenCount += 1;}
+
+eventEmitter.once('event-name', listener);
+eventEmitter.emit('event-name');  // listenCount == 1
+eventEmitter.emit('event-name');  // Ignored: listenCount == 1
+```
+
+### `on`
+
+```js
+import ApplicationEventEmitter from 'terra-application/lib/application-event-emitter';
+
+const eventEmitter = new ApplicationEventEmitter();
+let listenCount = 0;
+const listener = () => {listenCount += 1;}
+
+eventEmitter.on('event-name', listener);
+eventEmitter.emit('event-name');  // listenCount == 1
+eventEmitter.emit('event-name');  // listenCount == 2
+```
+
+### `off`
+
+```js
+import ApplicationEventEmitter from 'terra-application/lib/application-event-emitter';
+
+const eventEmitter = new ApplicationEventEmitter();
+let listenCount = 0;
+const listener = () => {listenCount += 1;}
+
+eventEmitter.on('event-name', listener);
+eventEmitter.on('event-name', listener);
```

Comment:
Should `eventEmitter.on('event-name', listener);` be listed twice in this example?

Reply:
I added twice here to illustrate that `off` removes all listeners registered for this event. I updated the `off` [API description and example](https://github.com/cerner/terra-application/pull/72/commits/845663e68f6bf6de6c0084464f88cbf26730bfce) to call out that it removes all listeners registered for this event. Do you think it's better to just list it once to minimize confusion?

Reply:
Yeah, i think it's a bit confusing, lets limit it to one.

Reply:
Sounds good. https://github.com/cerner/terra-application/pull/72/commits/6b1d1ee20b96eebe53a4eb985b16b8fb7406f973